source/features/subducting_plate_models/temperature/mass_conserving.cc

```diff
@@ -0,0 +1,338 @@
+            {
+
+              // 1. Determine initial heat content of the slab based on age of plate at trench
+              //    This uses the integral of the half-space temperature profile
+              const double initial_heat_content = 2*thermal_conductivity*(surface_temperature - potential_mantle_temperature) *
+                                                  std::sqrt(plate_age_sec/(thermal_diffusivity*const_pi));
+
+              //  2. Get Tmin and distance_offset given distance along slab and depth of point on the slab.
+              //  These equations are empirical based on fitting the temperature profiles from dynamic subduction models.
+              double min_temperature = 0.0;
+              double distance_offset = 0.0;
+
+              const double mantle_coupling_length = mantle_coupling_depth/std::sin(shallow_average_dip*const_pi/180.0); //m
+
+              /* Empirical model parameters */
+              double upper_mantle_depth = 660e3; // m
+              double upper_mantle_length = 1000e3; // m
+              /* Distance offset parameters */
+              double offset_distance_min = 25*1e3; // m
+              double offset_distance_max = 35*1e3; // m
+              double slope_distance_min = 25e3/(1.5*660e3);
+              double slope_distance_max = 20e3/(2.5*100e3);
+              double sink_velocity_min = 0.025; // m/yr
+              double sink_velocity_max = 0.10; // m/yr
+              /* minimum temperature parameters */
+              double slope_temperature_min = 250/(2.5*100*1e3); // deg/m (2.5 because dip in ref. model is 20-25 deg)
+              double slope_temperature_max = 500/(2.5*100*1e3); // deg/m
+              double temperature_min = 750;
+              double temperature_max = 950;
+              double vsubfact = (1 - (plate_velocity - sink_velocity_min)/sink_velocity_max);  // vsubfact = 0 when vel=sink_velocity_max
+
+              double slope_distance_shallow = slope_distance_min;
+              double slope_temperature_shallow = slope_temperature_min;
+
+              if (plate_velocity < sink_velocity_min)
+                {
+                  slope_distance_shallow = slope_distance_max;
+                  slope_temperature_shallow = slope_temperature_max;
+                }
+              else if ( plate_velocity > sink_velocity_max)
+                {
+                  slope_distance_shallow = slope_distance_min;
+                  slope_temperature_shallow = slope_temperature_min;
+                }
+              else
+                {
+                  slope_distance_shallow = slope_distance_min + vsubfact*(slope_distance_max - slope_distance_min);
+                  slope_temperature_shallow = slope_temperature_min + vsubfact*(slope_temperature_max - slope_temperature_min);
+                }
+              double offset_coupling_depth = slope_distance_shallow*mantle_coupling_depth; // m  mantle_coupling_length
```

Comment:
If variables are not changing anymore, can you mark them as `const`?

Reply:
I've made this change here and throughout the code.